setup.py

```diff
@@ -168,7 +168,7 @@ def package_files(prefix: str, directory: str, extension: str):
 
 extra_deps['nlp'] = [
     'transformers>=4.11,<5',
-    'datasets>=1.14,<2',
+    'datasets>=2.4.0,<3',
```

Comment:
shoulds this be >1.14 < 3?

and if it is >=2.4, we shoudl remove the `type: ignore`, and update the syntax to the datasets2 syntax?

Reply:
I removed the type: ignore and switched the name to `datasets.DownloadConfig`, which fixes the pyright error. I'm not sure what you mean by datasets2 syntax. Haven't really used `datasets` much but, in case this is relevant, [load_dataset](https://huggingface.co/docs/datasets/package_reference/loading_methods#datasets.load_dataset) seems to still be happy with the same API.

Reply:
Gonna call this resolved for the sake of getting this PR across the finish line. Happy to change something if you want though